presto-main/src/main/java/com/facebook/presto/execution/buffer/PagesSerdeFactory.java

```diff
@@ -25,19 +27,42 @@
 {
     private final BlockEncodingSerde blockEncodingSerde;
     private final boolean compressionEnabled;
+    private final boolean spillEncryptionEnabled;
 
-    public PagesSerdeFactory(BlockEncodingSerde blockEncodingSerde, boolean compressionEnabled)
+    public PagesSerdeFactory(BlockEncodingSerde blockEncodingSerde, boolean compressionEnabled, boolean spillEncryptionEnabled)
     {
         this.blockEncodingSerde = requireNonNull(blockEncodingSerde, "blockEncodingSerde is null");
         this.compressionEnabled = compressionEnabled;
+        this.spillEncryptionEnabled = spillEncryptionEnabled;
     }
 
     public PagesSerde createPagesSerde()
     {
+        return createPagesSerdeInternal(Optional.empty());
+    }
+
+    /**
+     * Creates a {@link PagesSerde} instance for use in spilling to disk, which will be encrypted if {@link PagesSerdeFactory#spillEncryptionEnabled}
+     * is set.
+     */
+    public PagesSerde createSpillerPagesSerde()
+    {
+        return createPagesSerdeInternal(spillEncryptionEnabled ? Optional.of(new AESSpillCipher(AESSpillCipher.Mode.AES_256)) : Optional.empty());
```

Comment:
consider introducing temp variable for readability and consistency with createPagesSerdeInternal:

```
        Optional<SpillCipher> spillCipher = Optional.empty();
        if (spillEncryptionEnabled) {
            spillCipher = Optional.of(new AesSpillCipher(AesSpillCipher.Mode.AES_256));
        }
        return createPagesSerdeInternal(spillCipher);
```